### DISCUSSION TITLE
Toggling DOM Inspector on/off adds a filter entry in the logger

### DISCUSSION THREAD

**Discussion by uBlock-user** [uBlockOrigin/uBlock-issues#1960] [General] [answered]
STR --

- Browse to https://github.com/gorhill/uBlock and open the logger in another tab. Refresh the github tab if needed.

- Click on enable DOM Inspector button and click on it again to disable.

- `@@||github.com^$ghide` appears in logger. 

- Now repeat step 2 in rapid succession and get spammed like below -- 

![Capture](https://user-images.githubusercontent.com/21290713/152361997-1122bae0-4eba-4f8e-bff4-ba4c3be1ab3a.PNG)

Firefox 98 and uBO dev build.

**Comment by gorhill** [comment]
It's because uBO adds an iframe to the page, to render its own overlay, and on the other side this causes an event that a new frame has been added and this causes an evaluation of generichide within that frame (because of `match_about_blank` in manifest), which is being reported to the logger. You get the same behavior if you launch the element picker. Not sure if it's fixable.

**Reply by uBlock-user** [reply]
Shall I mark this as wontfix then ?

**Comment by gorhill** [comment, marked as answer]
I doubt this can be fixed. Since I can also reproduce with element picker, in which the content scripts are not injected, this means there is a small window between the frame being added and the location of the frame being set to the element picker HTML, time window in which the frame is seen as a plain `about:blank` frame, meaning content scripts will be injected in it (because of `match_about_blank` in manifest), and hence reported to the logger. There is no way uBO's content script in an `about:blank` can tell whether the iframe is meant to be a container for the element picker (or DOM inspector overlay). But maybe further investigation can prove otherwise. I consider this a really minor issue and I don't plan to dedicate time to this.
